src/tournament.rs

```diff
@@ -66,6 +68,7 @@ pub struct Tournament {
 }
 
 impl Tournament {
+    #[no_mangle]
     pub extern fn from_preset(
```

Comment:
These look like they can be auto-generated to an extent by a build tool before compilation. Do you think this is worth looking into?

Reply:
I don't think it would be worth making a build script of this. The biggest reason is how we're handling errors. If `expected` is a `nullptr`, then something has gone wrong. Or some methods, exactly one thing can go wrong, so it can be inferred. For other methods, multiple things could go wrong so we need to return an error code.

I suppose we could write a couple of Rust proc macros, but even that seems like more work than we're saving.